extensions/blocks/image-compare/view.js

```diff
@@ -0,0 +1,559 @@
+			pageX = e.touches[ 0 ].pageX;
+		} else {
+			pageX = e.clientX + document.body.scrollLeft + document.documentElement.scrollLeft;
+		}
+		return pageX;
+	}
+
+	function getPageY( e ) {
+		let pageY;
+		if ( e.pageY ) {
+			pageY = e.pageY;
+		} else if ( e.touches ) {
+			pageY = e.touches[ 0 ].pageY;
+		} else {
+			pageY = e.clientY + document.body.scrollTop + document.documentElement.scrollTop;
+		}
+		return pageY;
+	}
+
+	function getLeftPercent( slider, input ) {
+		let leftPercent;
+		if ( typeof input === 'string' || typeof input === 'number' ) {
+			leftPercent = parseInt( input, 10 );
+		} else {
+			const sliderRect = slider.getBoundingClientRect();
+			const offset = {
+				top: sliderRect.top + document.body.scrollTop + document.documentElement.scrollTop,
+				left: sliderRect.left + document.body.scrollLeft + document.documentElement.scrollLeft,
+			};
+			const width = slider.offsetWidth;
+			const pageX = getPageX( input );
+			const relativeX = pageX - offset.left;
+			leftPercent = ( relativeX / width ) * 100;
+		}
+		return leftPercent;
+	}
+
+	function getTopPercent( slider, input ) {
+		let topPercent;
+		if ( typeof input === 'string' || typeof input === 'number' ) {
+			topPercent = parseInt( input, 10 );
+		} else {
+			const sliderRect = slider.getBoundingClientRect();
+			const offset = {
+				top: sliderRect.top + document.body.scrollTop + document.documentElement.scrollTop,
+				left: sliderRect.left + document.body.scrollLeft + document.documentElement.scrollLeft,
+			};
+			const width = slider.offsetHeight;
+			const pageY = getPageY( input );
+			const relativeY = pageY - offset.top;
+			topPercent = ( relativeY / width ) * 100;
+		}
+		return topPercent;
+	}
+
+	// values of BOOLEAN_OPTIONS are ignored. just used for 'in' test on keys
+	const BOOLEAN_OPTIONS = {
+		animate: true,
+		showLabels: true,
+		showCredits: true,
+		makeResponsive: true,
+	};
+	function interpretBoolean( x ) {
+		if ( typeof x !== 'string' ) {
+			return Boolean( x );
+		}
+		return ! ( x === 'false' || x === '' );
+	}
+
+	function JXSlider( selector, images, options ) {
+		this.selector = selector;
+
+		let i;
+		this.options = {
+			// new options must have default values set here.
+			animate: true,
+			showLabels: true,
+			showCredits: true,
+			makeResponsive: true,
+			startingPosition: '50%',
+			mode: 'horizontal',
+			callback: null, // pass a callback function if you like
+		};
+
+		for ( i in this.options ) {
+			if ( i in options ) {
+				if ( i in BOOLEAN_OPTIONS ) {
+					this.options[ i ] = interpretBoolean( options[ i ] );
+				} else {
+					this.options[ i ] = options[ i ];
+				}
+			}
+		}
+
+		if ( images.length === 2 ) {
+			this.imgBefore = new Graphic( images[ 0 ], this );
+			this.imgAfter = new Graphic( images[ 1 ], this );
+		}
+
+		if ( this.imgBefore.credit || this.imgAfter.credit ) {
+			this.options.showCredits = true;
+		} else {
+			this.options.showCredits = false;
+		}
+	}
```

Comment:
Are credits used anywhere in our implementation, or is this something we could get rid of?

Reply:
Good catch, removed.